### PR TITLE
Networking fixes

### LINKED ISSUE #939: Megazord releases don't include the BlahMegazord class
This breaks release 0.24.0

### PR DESCRIPTION
**Pull request by thomcc**
A few commits got dropped from the last build (guh), and while investigating them, I hit some other issues that I've fixed (ARM issue, and now we use newInput).

Fixes #939

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - `cargo test --all` produces no test failures
  - `cargo clippy --all --all-targets --all-features` runs without emitting any warnings
  - `cargo fmt` does not produce any changes to the code
  - `./gradlew ktlint detekt` runs without emitting any warnings
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
    - Testing was done manually.
- [x] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: N/A

### CODE REVIEW

File: components/viaduct/android/src/main/java/mozilla/appservices/httpconfig/HttpConfig.kt [hunk 1]
```diff
@@ -148,38 +147,3 @@ internal class CallbackImpl : RawFetchCallback {
     }
 }
 
-// The protobuf `bytes` type comes over as a com.google.protobuf.ByteString.
-// There's no provided way to convert/wrap this to an InputStream, so we do
-// that manually to avoid extra copying.
-internal class ByteStringInputStream(private val s: ByteString) : InputStream() {
```
Comment (eoger) [inline components/viaduct/android/src/main/java/mozilla/appservices/httpconfig/HttpConfig.kt]: Very good!

File: components/viaduct/src/backend/ffi.rs [hunk 2]
```diff
@@ -144,22 +144,32 @@ mod callback_holder {
     /// Note: We only assign to this once.
     static CALLBACK_PTR: AtomicUsize = AtomicUsize::new(0);
 
+    ffi_support::static_assert!(
```
Comment (eoger) [inline components/viaduct/src/backend/ffi.rs]: Pretty cool, a comment would be nice why we do these asserts

File: components/viaduct/src/backend/ffi.rs [hunk 3]
```diff
@@ -144,22 +144,32 @@ mod callback_holder {
...
-        } else {
-            Some(unsafe { *ptr })
-        }
+        let ptr_value = CALLBACK_PTR.load(Ordering::SeqCst);
+        unsafe { std::mem::transmute::<usize, Option<FetchCallback>>(ptr_value) }
     }
 
     /// Set the function pointer to the FetchCallback. Returns false if we did nothing because the callback had already been initialized
-    pub(super) unsafe fn set_callback(h: *const FetchCallback) -> bool {
-        let old_ptr = CALLBACK_PTR.compare_and_swap(0, h as usize, Ordering::SeqCst);
```
Comment (eoger) [inline components/viaduct/src/backend/ffi.rs]: Did the `as usize` cast fail on compile or during runtime? I'm not quite sure what the issue was about.

Comment (thomcc) [reply components/viaduct/src/backend/ffi.rs]: Hard crash when it calls. Not exactly sure why but something something ARM calling convention.

Comment (thomcc) [reply components/viaduct/src/backend/ffi.rs]: Or rather, I'm not exactly sure why it works on x86. This is the correct typing for this.

**Pull request merged**